src/lightning_classify/infer/infer.py

```diff
@@ -0,0 +1,106 @@
+import os
+import sys
+from pathlib import Path
+from typing import Dict, Optional, Tuple
+
+import numpy as np
+import torch
+from PIL import Image
+
+from lightning_classify.utils import downloader
+from lightning_classify.models import ImageRecogModelV1
+from lightning_classify.transforms import ImageTransform
+
+
+class ImageRecognition:
+    def __init__(self, device: torch.device, model_path=None):
+        
+        path_to_save = str(self._get_cache_dir()) + "/model.ckpt"
+        if model_path is None:
+            downloader(path_to_save)
+            self.model_path = path_to_save
+        else:
+            self.model_path = Path(model_path)
+            if not self.model_path.exists():
+                raise FileNotFoundError("Model does not exist, check your model location and read README for more information")
+        
+
+        self.device = torch.device(device)
+        self.class_dict = {
```

Comment:
dont make it manual, now i want you to save this thing inside the checkpoint or in the different file. let say we call it manifest.pth. In that file you will have anything related to the training info, one of them are class_dict.